### DISCUSSION TITLE
C/C++ Outline not working on Remote SSH machine

### DISCUSSION THREAD

**Discussion by deanhill1971** [microsoft/vscode-cpptools#11620] [General]
My local machine is Windows 10 (VS Code 1.83.1, C/C++ Extension 1.18.0). 
My remote machine is Red Hat 8 (VS Code 1.81.1, C/C++ Extension 1.18.0).
Using VS Code locally on either of these machines to view C++ files works well, e.g. Outline and Go To Definition work fine.

But the response time is slow when running locally on the RH machine, so I setup VS Code Remote SSH.

The connection from Windows to RH works well and the response time is much better. 
The problem is that when remoted into RH the Outline and Go To Definition features don't work.
When doing the remote and going to the Extensions pane, I see LOCAL - INSTALLED and SSH: <machine-name> - INSTALLED. Under LOCAL there is C/C++ 1.18.0. Under SHH there is C/C++ 1.18.1 (notice the last .1).

In the lower right when I click "{}" it shows IntelliSense: Ready, (a barrel icon) Initializing Workspace, Code Analysis Mode: Manual.

Security on the RH machine is fairly tight, so I had to manually install the C/C++ extension on that box from a vsix file which I think is how I ended up with 1.18.1 there.

Any thoughts on how to get C++ parsing working when doing Remote SSH?

**Comment by starball5** [comment]
Related on Stack Overflow: [VS Code C/C++ IntelliSense not working with Remote SSH](https://stackoverflow.com/q/77419276/11107541)

**Comment by deanhill1971** [comment]
I found a solution and documented it as the accepted answer on Stack Overflow: https://stackoverflow.com/questions/77419276/vs-code-c-c-intellisense-not-working-with-remote-ssh
